feat(backend): don't require authentication for /get-released-data

corneliusroemer opened the pull request:
resolves #1913
resolves #1625

preview URL: https://get-no-auth.loculus.org

### Summary

- Don't require authn/authz for `get-released-data` endpoint
- Fix lints
- Run backend tests in paralell

fengelniederhammer commented on backend/src/main/kotlin/org/loculus/backend/config/SecurityConfig.kt:
```diff
@@ -4,7 +4,6 @@ import jakarta.servlet.http.HttpServletRequest
 import jakarta.servlet.http.HttpServletResponse
 import mu.KotlinLogging
 import org.loculus.backend.auth.Roles.EXTERNAL_METADATA_UPDATER
-import org.loculus.backend.auth.Roles.GET_RELEASED_DATA
```
The `GET_RELEASED_DATA` role can probably deleleted, too?

corneliusroemer replied:
Done

corneliusroemer commented:
These are the failures, at least locally:

<details>

```
> Task :test

DeleteAllSequenceDataEndpointTest > GIVEN I deleted all sequences WHEN submitting again THEN new sequences are present() FAILED
    java.lang.AssertionError: Status expected:<200> but was:<401>
        at org.springframework.test.util.AssertionErrors.fail(AssertionErrors.java:59)
        at org.springframework.test.util.AssertionErrors.assertEquals(AssertionErrors.java:122)
        at org.springframework.test.web.servlet.result.StatusResultMatchers.lambda$matcher$9(StatusResultMatchers.java:637)
        at org.springframework.test.web.servlet.MockMvc$1.andExpect(MockMvc.java:214)
        at org.loculus.backend.controller.submission.SubmissionConvenienceClient.getReleasedData(SubmissionConvenienceClient.kt:590)
        at org.loculus.backend.controller.submission.SubmissionConvenienceClient.getReleasedData$default(SubmissionConvenienceClient.kt:382)
        at org.loculus.backend.controller.debug.DeleteAllSequenceDataEndpointTest.GIVEN I deleted all sequences WHEN submitting again THEN new sequences are present(DeleteAllSequenceDataEndpointTest.kt:149)

DeleteAllSequenceDataEndpointTest > GIVEN released sequences WHEN deleting all sequences THEN has no released sequences() FAILED
    java.lang.AssertionError: Status expected:<200> but was:<401>
        at org.springframework.test.util.AssertionErrors.fail(AssertionErrors.java:59)
        at org.springframework.test.util.AssertionErrors.assertEquals(AssertionErrors.java:122)
        at org.springframework.test.web.servlet.result.StatusResultMatchers.lambda$matcher$9(StatusResultMatchers.java:637)
        at org.springframework.test.web.servlet.MockMvc$1.andExpect(MockMvc.java:214)
        at org.loculus.backend.controller.submission.SubmissionConvenienceClient.getReleasedData(SubmissionConvenienceClient.kt:590)
        at org.loculus.backend.controller.submission.SubmissionConvenienceClient.getReleasedData$default(SubmissionConvenienceClient.kt:382)
        at org.loculus.backend.controller.debug.DeleteAllSequenceDataEndpointTest.GIVEN released sequences WHEN deleting all sequences THEN has no released sequences(DeleteAllSequenceDataEndpointTest.kt:68)

GetReleasedDataEndpointTest > GIVEN released data exists THEN returns it with additional metadata fields() FAILED
    java.lang.AssertionError: Status expected:<200> but was:<401>
        at org.springframework.test.util.AssertionErrors.fail(AssertionErrors.java:59)
        at org.springframework.test.util.AssertionErrors.assertEquals(AssertionErrors.java:122)
        at org.springframework.test.web.servlet.result.StatusResultMatchers.lambda$matcher$9(StatusResultMatchers.java:637)
        at org.springframework.test.web.servlet.MockMvc$1.andExpect(MockMvc.java:214)
        at org.loculus.backend.controller.submission.GetReleasedDataEndpointTest.GIVEN released data exists THEN returns it with additional metadata fields(GetReleasedDataEndpointTest.kt:335)

GetReleasedDataEndpointTest > GIVEN no sequence entries in database THEN returns empty response() FAILED
    java.lang.AssertionError: Status expected:<200> but was:<401>
        at org.springframework.test.util.AssertionErrors.fail(AssertionErrors.java:59)
        at org.springframework.test.util.AssertionErrors.assertEquals(AssertionErrors.java:122)
        at org.springframework.test.web.servlet.result.StatusResultMatchers.lambda$matcher$9(StatusResultMatchers.java:637)
        at org.springframework.test.web.servlet.MockMvc$1.andExpect(MockMvc.java:214)
        at org.loculus.backend.controller.submission.GetReleasedDataEndpointTest.GIVEN no sequence entries in database THEN returns empty response(GetReleasedDataEndpointTest.kt:313)

GetReleasedDataEndpointTest > GIVEN preprocessing pipeline submitted with missing metadata fields THEN fields are filled with null() FAILED
    java.lang.AssertionError: Status expected:<200> but was:<401>
        at org.springframework.test.util.AssertionErrors.fail(AssertionErrors.java:59)
        at org.springframework.test.util.AssertionErrors.assertEquals(AssertionErrors.java:122)
        at org.springframework.test.web.servlet.result.StatusResultMatchers.lambda$matcher$9(StatusResultMatchers.java:637)
        at org.springframework.test.web.servlet.MockMvc$1.andExpect(MockMvc.java:214)
        at org.loculus.backend.controller.submission.GetReleasedDataEndpointTest.GIVEN preprocessing pipeline submitted with missing metadata fields THEN fields are filled with null(GetReleasedDataEndpointTest.kt:385)

GetReleasedDataEndpointTest > GIVEN released data exists in multiple versions THEN the 'versionStatus' flag is set correctly() FAILED
    java.lang.AssertionError: Status expected:<200> but was:<401>
        at org.springframework.test.util.AssertionErrors.fail(AssertionErrors.java:59)
        at org.springframework.test.util.AssertionErrors.assertEquals(AssertionErrors.java:122)
        at org.springframework.test.web.servlet.result.StatusResultMatchers.lambda$matcher$9(StatusResultMatchers.java:637)
        at org.springframework.test.web.servlet.MockMvc$1.andExpect(MockMvc.java:214)
        at org.loculus.backend.controller.submission.GetReleasedDataEndpointTest.GIVEN released data exists in multiple versions THEN the 'versionStatus' flag is set correctly(GetReleasedDataEndpointTest.kt:359)

GetReleasedDataEndpointTest > WHEN I request zstd compressed data THEN should return zstd compressed data() FAILED
    java.lang.AssertionError: Status expected:<200> but was:<401>
        at org.springframework.test.util.AssertionErrors.fail(AssertionErrors.java:59)
        at org.springframework.test.util.AssertionErrors.assertEquals(AssertionErrors.java:122)
        at org.springframework.test.web.servlet.result.StatusResultMatchers.lambda$matcher$9(StatusResultMatchers.java:637)
        at org.springframework.test.web.servlet.MockMvc$1.andExpect(MockMvc.java:214)
        at org.loculus.backend.controller.submission.GetReleasedDataEndpointTest.WHEN I request zstd compressed data THEN should return zstd compressed data(GetReleasedDataEndpointTest.kt:243)

GetReleasedDataEndpointTest > GIVEN revocation version THEN all data is present but mostly null() FAILED
    java.lang.AssertionError: Status expected:<200> but was:<401>
        at org.springframework.test.util.AssertionErrors.fail(AssertionErrors.java:59)
        at org.springframework.test.util.AssertionErrors.assertEquals(AssertionErrors.java:122)
        at org.springframework.test.web.servlet.result.StatusResultMatchers.lambda$matcher$9(StatusResultMatchers.java:637)
        at org.springframework.test.web.servlet.MockMvc$1.andExpect(MockMvc.java:214)
        at org.loculus.backend.controller.submission.GetReleasedDataEndpointTest.GIVEN revocation version THEN all data is present but mostly null(GetReleasedDataEndpointTest.kt:407)

SubmissionJourneyTest > Release journey scenario for two organisms() FAILED
    java.lang.AssertionError: Status expected:<200> but was:<401>
        at org.springframework.test.util.AssertionErrors.fail(AssertionErrors.java:59)
        at org.springframework.test.util.AssertionErrors.assertEquals(AssertionErrors.java:122)
        at org.springframework.test.web.servlet.result.StatusResultMatchers.lambda$matcher$9(StatusResultMatchers.java:637)
        at org.springframework.test.web.servlet.MockMvc$1.andExpect(MockMvc.java:214)
        at org.loculus.backend.controller.submission.SubmissionConvenienceClient.getReleasedData(SubmissionConvenienceClient.kt:590)
        at org.loculus.backend.controller.submission.SubmissionJourneyTest.Release journey scenario for two organisms(SubmissionJourneyTest.kt:144)

SubmitExternalMetadataEndpointTest > GIVEN accessions do not exist THEN do not allow submission() FAILED
    java.lang.AssertionError: Status expected:<200> but was:<401>
        at org.springframework.test.util.AssertionErrors.fail(AssertionErrors.java:59)
        at org.springframework.test.util.AssertionErrors.assertEquals(AssertionErrors.java:122)
        at org.springframework.test.web.servlet.result.StatusResultMatchers.lambda$matcher$9(StatusResultMatchers.java:637)
        at org.springframework.test.web.servlet.MockMvc$1.andExpect(MockMvc.java:214)
        at org.loculus.backend.controller.submission.SubmissionConvenienceClient.getReleasedData(SubmissionConvenienceClient.kt:590)
        at org.loculus.backend.controller.submission.SubmissionConvenienceClient.getReleasedData$default(SubmissionConvenienceClient.kt:382)
        at org.loculus.backend.controller.submission.SubmitExternalMetadataEndpointTest.GIVEN accessions do not exist THEN do not allow submission(SubmitExternalMetadataEndpointTest.kt:161)

SubmitExternalMetadataEndpointTest > WHEN I add a metadata field of another external updater THEN returns unprocessable entity() FAILED
    java.lang.AssertionError: Status expected:<200> but was:<401>
        at org.springframework.test.util.AssertionErrors.fail(AssertionErrors.java:59)
        at org.springframework.test.util.AssertionErrors.assertEquals(AssertionErrors.java:122)
        at org.springframework.test.web.servlet.result.StatusResultMatchers.lambda$matcher$9(StatusResultMatchers.java:637)
        at org.springframework.test.web.servlet.MockMvc$1.andExpect(MockMvc.java:214)
        at org.loculus.backend.controller.submission.SubmissionConvenienceClient.getReleasedData(SubmissionConvenienceClient.kt:590)
        at org.loculus.backend.controller.submission.SubmissionConvenienceClient.getReleasedData$default(SubmissionConvenienceClient.kt:382)
        at org.loculus.backend.controller.submission.SubmitExternalMetadataEndpointTest.WHEN I add a metadata field of another external updater THEN returns unprocessable entity(SubmitExternalMetadataEndpointTest.kt:110)

SubmitExternalMetadataEndpointTest > GIVEN accessions are in status released THEN add external metadata() FAILED
    java.lang.AssertionError: Status expected:<200> but was:<401>
        at org.springframework.test.util.AssertionErrors.fail(AssertionErrors.java:59)
        at org.springframework.test.util.AssertionErrors.assertEquals(AssertionErrors.java:122)
        at org.springframework.test.web.servlet.result.StatusResultMatchers.lambda$matcher$9(StatusResultMatchers.java:637)
        at org.springframework.test.web.servlet.MockMvc$1.andExpect(MockMvc.java:214)
        at org.loculus.backend.controller.submission.SubmissionConvenienceClient.getReleasedData(SubmissionConvenienceClient.kt:590)
        at org.loculus.backend.controller.submission.SubmissionConvenienceClient.getReleasedData$default(SubmissionConvenienceClient.kt:382)
        at org.loculus.backend.controller.submission.SubmitExternalMetadataEndpointTest.GIVEN accessions are in status released THEN add external metadata(SubmitExternalMetadataEndpointTest.kt:60)

SubmitExternalMetadataEndpointTest > GIVEN accessions are in status released THEN add external metadata from multiple sources() FAILED
    java.lang.AssertionError: Status expected:<200> but was:<401>
        at org.springframework.test.util.AssertionErrors.fail(AssertionErrors.java:59)
        at org.springframework.test.util.AssertionErrors.assertEquals(AssertionErrors.java:122)
        at org.springframework.test.web.servlet.result.StatusResultMatchers.lambda$matcher$9(StatusResultMatchers.java:637)
        at org.springframework.test.web.servlet.MockMvc$1.andExpect(MockMvc.java:214)
        at org.loculus.backend.controller.submission.SubmissionConvenienceClient.getReleasedData(SubmissionConvenienceClient.kt:590)
        at org.loculus.backend.controller.submission.SubmissionConvenienceClient.getReleasedData$default(SubmissionConvenienceClient.kt:382)
        at org.loculus.backend.controller.submission.SubmitExternalMetadataEndpointTest.GIVEN accessions are in status released THEN add external metadata from multiple sources(SubmitExternalMetadataEndpointTest.kt:86)

SubmitExternalMetadataEndpointTest > GIVEN accessions are not yet in status released THEN do not allow submission() FAILED
    java.lang.AssertionError: Status expected:<200> but was:<401>
        at org.springframework.test.util.AssertionErrors.fail(AssertionErrors.java:59)
        at org.springframework.test.util.AssertionErrors.assertEquals(AssertionErrors.java:122)
        at org.springframework.test.web.servlet.result.StatusResultMatchers.lambda$matcher$9(StatusResultMatchers.java:637)
        at org.springframework.test.web.servlet.MockMvc$1.andExpect(MockMvc.java:214)
        at org.loculus.backend.controller.submission.SubmissionConvenienceClient.getReleasedData(SubmissionConvenienceClient.kt:590)
        at org.loculus.backend.controller.submission.SubmissionConvenienceClient.getReleasedData$default(SubmissionConvenienceClient.kt:382)
        at org.loculus.backend.controller.submission.SubmitExternalMetadataEndpointTest.GIVEN accessions are not yet in status released THEN do not allow submission(SubmitExternalMetadataEndpointTest.kt:137)

ApproveProcessedDataEndpointTest > GIVEN revoked sequence entries awaiting approval THEN their status should be APPROVED_FOR_RELEASE() FAILED
    java.lang.AssertionError: JSON path "$[*].accession" expected:<[LOC_000001Y, LOC_000002W, LOC_000003U, LOC_000004S, LOC_000005Q, LOC_000006N, LOC_000007L, LOC_000008J, LOC_000009G, LOC_00000AE]> but was:<["LOC_00000AE","LOC_000008J","LOC_000001Y","LOC_000003U","LOC_000002W","LOC_000006N","LOC_000004S","LOC_000005Q","LOC_000009G","LOC_000007L"]>
        at org.springframework.test.util.AssertionErrors.fail(AssertionErrors.java:59)
        at org.springframework.test.util.AssertionErrors.assertEquals(AssertionErrors.java:122)
        at org.springframework.test.util.JsonPathExpectationsHelper.assertValue(JsonPathExpectationsHelper.java:123)
        at org.springframework.test.web.servlet.result.JsonPathResultMatchers.lambda$value$2(JsonPathResultMatchers.java:111)
        at org.springframework.test.web.servlet.MockMvc$1.andExpect(MockMvc.java:214)
        at org.loculus.backend.controller.submission.ApproveProcessedDataEndpointTest.GIVEN revoked sequence entries awaiting approval THEN their status should be APPROVED_FOR_RELEASE(ApproveProcessedDataEndpointTest.kt:70)

DeleteSequencesEndpointTest > WHEN superuser deletes all entries THEN is successfully deleted() FAILED
    java.lang.AssertionError: JSON path "$[0].accession" expected:<LOC_000001Y> but was:<LOC_000006N>
        at org.springframework.test.util.AssertionErrors.fail(AssertionErrors.java:59)
        at org.springframework.test.util.AssertionErrors.assertEquals(AssertionErrors.java:122)
        at org.springframework.test.util.JsonPathExpectationsHelper.assertValue(JsonPathExpectationsHelper.java:123)
        at org.springframework.test.web.servlet.result.JsonPathResultMatchers.lambda$value$2(JsonPathResultMatchers.java:111)
        at org.springframework.test.web.servlet.MockMvc$1.andExpect(MockMvc.java:214)
        at org.loculus.backend.controller.submission.DeleteSequencesEndpointTest.WHEN superuser deletes all entries THEN is successfully deleted(DeleteSequencesEndpointTest.kt:288)

DeleteSequencesEndpointTest > WHEN superuser deletes entries of other user THEN is successfully deleted() FAILED
    java.lang.AssertionError: JSON path "$[0].accession" expected:<LOC_000001Y> but was:<LOC_00000AE>
        at org.springframework.test.util.AssertionErrors.fail(AssertionErrors.java:59)
        at org.springframework.test.util.AssertionErrors.assertEquals(AssertionErrors.java:122)
        at org.springframework.test.util.JsonPathExpectationsHelper.assertValue(JsonPathExpectationsHelper.java:123)
        at org.springframework.test.web.servlet.result.JsonPathResultMatchers.lambda$value$2(JsonPathResultMatchers.java:111)
        at org.springframework.test.web.servlet.MockMvc$1.andExpect(MockMvc.java:214)
        at org.loculus.backend.controller.submission.DeleteSequencesEndpointTest.WHEN superuser deletes entries of other user THEN is successfully deleted(DeleteSequencesEndpointTest.kt:308)

292 tests completed, 17 failed

> Task :test FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':test'.
> There were failing tests. See the report at: file:///Users/corneliusromer/code/loculus/backend/build/reports/tests/test/index.html

* Try:
> Run with --scan to get full insights.

BUILD FAILED in 1m 12s
5 actionable tasks: 1 executed, 4 up-to-date

loculus/backend on  get-no-auth [$] via ☕ v21.0.4 via 🅶 v8.5 in ☸ aws (traefik) on ☁️  loculus.eks (eu-central-1) took 1m13s
❌1 ❯ docker stop loculus_postgres
loculus_postgres

loculus/backend on  get-no-auth [$] via ☕ v21.0.4 via 🅶 v8.5 in ☸ aws (traefik) on ☁️  loculus.eks (eu-central-1)
❯    docker run -d \
   --name loculus_postgres \
   -e POSTGRES_DB=loculus \
   -e POSTGRES_USER=postgres \
   -e POSTGRES_PASSWORD=unsecure \
   -p 5432:5432 \
   postgres:latest
docker: Error response from daemon: Conflict. The container name "/loculus_postgres" is already in use by container "68098de8d8492631a4402873a89b2e56fe9c9703a0c603499ba2b30fdc6e1a8a". You have to remove (or rename) that container to be able to reuse that name.
See 'docker run --help'.

loculus/backend on  get-no-auth [$] via ☕ v21.0.4 via 🅶 v8.5 in ☸ aws (traefik) on ☁️  loculus.eks (eu-central-1)
❌125 ❯ docker rm loculus_postgres
loculus_postgres

loculus/backend on  get-no-auth [$] via ☕ v21.0.4 via 🅶 v8.5 in ☸ aws (traefik) on ☁️  loculus.eks (eu-central-1)
❯    docker run -d \
   --name loculus_postgres \
   -e POSTGRES_DB=loculus \
   -e POSTGRES_USER=postgres \
   -e POSTGRES_PASSWORD=unsecure \
   -p 5432:5432 \
   postgres:latest
f37e2a3c68dc26026ca33ba45bc82f29bd1661694f07c9384f870a2c11064711

loculus/backend on  get-no-auth [$] via ☕ v21.0.4 via 🅶 v8.5 in ☸ aws (traefik) on ☁️  loculus.eks (eu-central-1)
❯ ./gradlew test

> Task :test

DeleteAllSequenceDataEndpointTest > GIVEN I deleted all sequences WHEN submitting again THEN new sequences are present() FAILED
    java.lang.AssertionError: Status expected:<200> but was:<401>
        at org.springframework.test.util.AssertionErrors.fail(AssertionErrors.java:59)
        at org.springframework.test.util.AssertionErrors.assertEquals(AssertionErrors.java:122)
        at org.springframework.test.web.servlet.result.StatusResultMatchers.lambda$matcher$9(StatusResultMatchers.java:637)
        at org.springframework.test.web.servlet.MockMvc$1.andExpect(MockMvc.java:214)
        at org.loculus.backend.controller.submission.SubmissionConvenienceClient.getReleasedData(SubmissionConvenienceClient.kt:590)
        at org.loculus.backend.controller.submission.SubmissionConvenienceClient.getReleasedData$default(SubmissionConvenienceClient.kt:382)
        at org.loculus.backend.controller.debug.DeleteAllSequenceDataEndpointTest.GIVEN I deleted all sequences WHEN submitting again THEN new sequences are present(DeleteAllSequenceDataEndpointTest.kt:149)

DeleteAllSequenceDataEndpointTest > GIVEN released sequences WHEN deleting all sequences THEN has no released sequences() FAILED
    java.lang.AssertionError: Status expected:<200> but was:<401>
        at org.springframework.test.util.AssertionErrors.fail(AssertionErrors.java:59)
        at org.springframework.test.util.AssertionErrors.assertEquals(AssertionErrors.java:122)
        at org.springframework.test.web.servlet.result.StatusResultMatchers.lambda$matcher$9(StatusResultMatchers.java:637)
        at org.springframework.test.web.servlet.MockMvc$1.andExpect(MockMvc.java:214)
        at org.loculus.backend.controller.submission.SubmissionConvenienceClient.getReleasedData(SubmissionConvenienceClient.kt:590)
        at org.loculus.backend.controller.submission.SubmissionConvenienceClient.getReleasedData$default(SubmissionConvenienceClient.kt:382)
        at org.loculus.backend.controller.debug.DeleteAllSequenceDataEndpointTest.GIVEN released sequences WHEN deleting all sequences THEN has no released sequences(DeleteAllSequenceDataEndpointTest.kt:68)

GetReleasedDataEndpointTest > GIVEN released data exists THEN returns it with additional metadata fields() FAILED
    java.lang.AssertionError: Status expected:<200> but was:<401>
        at org.springframework.test.util.AssertionErrors.fail(AssertionErrors.java:59)
        at org.springframework.test.util.AssertionErrors.assertEquals(AssertionErrors.java:122)
        at org.springframework.test.web.servlet.result.StatusResultMatchers.lambda$matcher$9(StatusResultMatchers.java:637)
        at org.springframework.test.web.servlet.MockMvc$1.andExpect(MockMvc.java:214)
        at org.loculus.backend.controller.submission.GetReleasedDataEndpointTest.GIVEN released data exists THEN returns it with additional metadata fields(GetReleasedDataEndpointTest.kt:335)

GetReleasedDataEndpointTest > GIVEN no sequence entries in database THEN returns empty response() FAILED
    java.lang.AssertionError: Status expected:<200> but was:<401>
        at org.springframework.test.util.AssertionErrors.fail(AssertionErrors.java:59)
        at org.springframework.test.util.AssertionErrors.assertEquals(AssertionErrors.java:122)
        at org.springframework.test.web.servlet.result.StatusResultMatchers.lambda$matcher$9(StatusResultMatchers.java:637)
        at org.springframework.test.web.servlet.MockMvc$1.andExpect(MockMvc.java:214)
        at org.loculus.backend.controller.submission.GetReleasedDataEndpointTest.GIVEN no sequence entries in database THEN returns empty response(GetReleasedDataEndpointTest.kt:313)

GetReleasedDataEndpointTest > GIVEN preprocessing pipeline submitted with missing metadata fields THEN fields are filled with null() FAILED
    java.lang.AssertionError: Status expected:<200> but was:<401>
        at org.springframework.test.util.AssertionErrors.fail(AssertionErrors.java:59)
        at org.springframework.test.util.AssertionErrors.assertEquals(AssertionErrors.java:122)
        at org.springframework.test.web.servlet.result.StatusResultMatchers.lambda$matcher$9(StatusResultMatchers.java:637)
        at org.springframework.test.web.servlet.MockMvc$1.andExpect(MockMvc.java:214)
        at org.loculus.backend.controller.submission.GetReleasedDataEndpointTest.GIVEN preprocessing pipeline submitted with missing metadata fields THEN fields are filled with null(GetReleasedDataEndpointTest.kt:385)

GetReleasedDataEndpointTest > GIVEN released data exists in multiple versions THEN the 'versionStatus' flag is set correctly() FAILED
    java.lang.AssertionError: Status expected:<200> but was:<401>
        at org.springframework.test.util.AssertionErrors.fail(AssertionErrors.java:59)
        at org.springframework.test.util.AssertionErrors.assertEquals(AssertionErrors.java:122)
        at org.springframework.test.web.servlet.result.StatusResultMatchers.lambda$matcher$9(StatusResultMatchers.java:637)
        at org.springframework.test.web.servlet.MockMvc$1.andExpect(MockMvc.java:214)
        at org.loculus.backend.controller.submission.GetReleasedDataEndpointTest.GIVEN released data exists in multiple versions THEN the 'versionStatus' flag is set correctly(GetReleasedDataEndpointTest.kt:359)

GetReleasedDataEndpointTest > WHEN I request zstd compressed data THEN should return zstd compressed data() FAILED
    java.lang.AssertionError: Status expected:<200> but was:<401>
        at org.springframework.test.util.AssertionErrors.fail(AssertionErrors.java:59)
        at org.springframework.test.util.AssertionErrors.assertEquals(AssertionErrors.java:122)
        at org.springframework.test.web.servlet.result.StatusResultMatchers.lambda$matcher$9(StatusResultMatchers.java:637)
        at org.springframework.test.web.servlet.MockMvc$1.andExpect(MockMvc.java:214)
        at org.loculus.backend.controller.submission.GetReleasedDataEndpointTest.WHEN I request zstd compressed data THEN should return zstd compressed data(GetReleasedDataEndpointTest.kt:243)

GetReleasedDataEndpointTest > GIVEN revocation version THEN all data is present but mostly null() FAILED
    java.lang.AssertionError: Status expected:<200> but was:<401>
        at org.springframework.test.util.AssertionErrors.fail(AssertionErrors.java:59)
        at org.springframework.test.util.AssertionErrors.assertEquals(AssertionErrors.java:122)
        at org.springframework.test.web.servlet.result.StatusResultMatchers.lambda$matcher$9(StatusResultMatchers.java:637)
        at org.springframework.test.web.servlet.MockMvc$1.andExpect(MockMvc.java:214)
        at org.loculus.backend.controller.submission.GetReleasedDataEndpointTest.GIVEN revocation version THEN all data is present but mostly null(GetReleasedDataEndpointTest.kt:407)

SubmissionJourneyTest > Release journey scenario for two organisms() FAILED
    java.lang.AssertionError: Status expected:<200> but was:<401>
        at org.springframework.test.util.AssertionErrors.fail(AssertionErrors.java:59)
        at org.springframework.test.util.AssertionErrors.assertEquals(AssertionErrors.java:122)
        at org.springframework.test.web.servlet.result.StatusResultMatchers.lambda$matcher$9(StatusResultMatchers.java:637)
        at org.springframework.test.web.servlet.MockMvc$1.andExpect(MockMvc.java:214)
        at org.loculus.backend.controller.submission.SubmissionConvenienceClient.getReleasedData(SubmissionConvenienceClient.kt:590)
        at org.loculus.backend.controller.submission.SubmissionJourneyTest.Release journey scenario for two organisms(SubmissionJourneyTest.kt:144)

SubmitExternalMetadataEndpointTest > GIVEN accessions do not exist THEN do not allow submission() FAILED
    java.lang.AssertionError: Status expected:<200> but was:<401>
        at org.springframework.test.util.AssertionErrors.fail(AssertionErrors.java:59)
        at org.springframework.test.util.AssertionErrors.assertEquals(AssertionErrors.java:122)
        at org.springframework.test.web.servlet.result.StatusResultMatchers.lambda$matcher$9(StatusResultMatchers.java:637)
        at org.springframework.test.web.servlet.MockMvc$1.andExpect(MockMvc.java:214)
        at org.loculus.backend.controller.submission.SubmissionConvenienceClient.getReleasedData(SubmissionConvenienceClient.kt:590)
        at org.loculus.backend.controller.submission.SubmissionConvenienceClient.getReleasedData$default(SubmissionConvenienceClient.kt:382)
        at org.loculus.backend.controller.submission.SubmitExternalMetadataEndpointTest.GIVEN accessions do not exist THEN do not allow submission(SubmitExternalMetadataEndpointTest.kt:161)

SubmitExternalMetadataEndpointTest > WHEN I add a metadata field of another external updater THEN returns unprocessable entity() FAILED
    java.lang.AssertionError: Status expected:<200> but was:<401>
        at org.springframework.test.util.AssertionErrors.fail(AssertionErrors.java:59)
        at org.springframework.test.util.AssertionErrors.assertEquals(AssertionErrors.java:122)
        at org.springframework.test.web.servlet.result.StatusResultMatchers.lambda$matcher$9(StatusResultMatchers.java:637)
        at org.springframework.test.web.servlet.MockMvc$1.andExpect(MockMvc.java:214)
        at org.loculus.backend.controller.submission.SubmissionConvenienceClient.getReleasedData(SubmissionConvenienceClient.kt:590)
        at org.loculus.backend.controller.submission.SubmissionConvenienceClient.getReleasedData$default(SubmissionConvenienceClient.kt:382)
        at org.loculus.backend.controller.submission.SubmitExternalMetadataEndpointTest.WHEN I add a metadata field of another external updater THEN returns unprocessable entity(SubmitExternalMetadataEndpointTest.kt:110)

SubmitExternalMetadataEndpointTest > GIVEN accessions are in status released THEN add external metadata() FAILED
    java.lang.AssertionError: Status expected:<200> but was:<401>
        at org.springframework.test.util.AssertionErrors.fail(AssertionErrors.java:59)
        at org.springframework.test.util.AssertionErrors.assertEquals(AssertionErrors.java:122)
        at org.springframework.test.web.servlet.result.StatusResultMatchers.lambda$matcher$9(StatusResultMatchers.java:637)
        at org.springframework.test.web.servlet.MockMvc$1.andExpect(MockMvc.java:214)
        at org.loculus.backend.controller.submission.SubmissionConvenienceClient.getReleasedData(SubmissionConvenienceClient.kt:590)
        at org.loculus.backend.controller.submission.SubmissionConvenienceClient.getReleasedData$default(SubmissionConvenienceClient.kt:382)
        at org.loculus.backend.controller.submission.SubmitExternalMetadataEndpointTest.GIVEN accessions are in status released THEN add external metadata(SubmitExternalMetadataEndpointTest.kt:60)

SubmitExternalMetadataEndpointTest > GIVEN accessions are in status released THEN add external metadata from multiple sources() FAILED
    java.lang.AssertionError: Status expected:<200> but was:<401>
        at org.springframework.test.util.AssertionErrors.fail(AssertionErrors.java:59)
        at org.springframework.test.util.AssertionErrors.assertEquals(AssertionErrors.java:122)
        at org.springframework.test.web.servlet.result.StatusResultMatchers.lambda$matcher$9(StatusResultMatchers.java:637)
        at org.springframework.test.web.servlet.MockMvc$1.andExpect(MockMvc.java:214)
        at org.loculus.backend.controller.submission.SubmissionConvenienceClient.getReleasedData(SubmissionConvenienceClient.kt:590)
        at org.loculus.backend.controller.submission.SubmissionConvenienceClient.getReleasedData$default(SubmissionConvenienceClient.kt:382)
        at org.loculus.backend.controller.submission.SubmitExternalMetadataEndpointTest.GIVEN accessions are in status released THEN add external metadata from multiple sources(SubmitExternalMetadataEndpointTest.kt:86)

SubmitExternalMetadataEndpointTest > GIVEN accessions are not yet in status released THEN do not allow submission() FAILED
    java.lang.AssertionError: Status expected:<200> but was:<401>
        at org.springframework.test.util.AssertionErrors.fail(AssertionErrors.java:59)
        at org.springframework.test.util.AssertionErrors.assertEquals(AssertionErrors.java:122)
        at org.springframework.test.web.servlet.result.StatusResultMatchers.lambda$matcher$9(StatusResultMatchers.java:637)
        at org.springframework.test.web.servlet.MockMvc$1.andExpect(MockMvc.java:214)
        at org.loculus.backend.controller.submission.SubmissionConvenienceClient.getReleasedData(SubmissionConvenienceClient.kt:590)
        at org.loculus.backend.controller.submission.SubmissionConvenienceClient.getReleasedData$default(SubmissionConvenienceClient.kt:382)
        at org.loculus.backend.controller.submission.SubmitExternalMetadataEndpointTest.GIVEN accessions are not yet in status released THEN do not allow submission(SubmitExternalMetadataEndpointTest.kt:137)
```

</details>
